src/mavsdk/plugins/camera/camera_impl.cpp

```diff
@@ -1417,274 +1250,320 @@ void CameraImpl::process_camera_information(const mavlink_message_t& message)
     camera_information.model_name[sizeof(camera_information.model_name) - 1] = '\0';
     camera_information.cam_definition_uri[sizeof(camera_information.cam_definition_uri) - 1] = '\0';
 
-    std::lock_guard<std::mutex> lock(_information.mutex);
-
-    _information.data.vendor_name = (char*)(camera_information.vendor_name);
-    _information.data.model_name = (char*)(camera_information.model_name);
-    _information.data.focal_length_mm = camera_information.focal_length;
-    _information.data.horizontal_sensor_size_mm = camera_information.sensor_size_h;
-    _information.data.vertical_sensor_size_mm = camera_information.sensor_size_v;
-    _information.data.horizontal_resolution_px = camera_information.resolution_h;
-    _information.data.vertical_resolution_px = camera_information.resolution_v;
-
-    _information.subscription_callbacks.queue(
-        _information.data, [this](const auto& func) { _system_impl->call_user_callback(func); });
+    Camera::Information new_information{};
+    // TODO: Check the case for 1-6.
```

Comment:
What does that mean?

Reply:
Oh thanks, yes I should fix that. 1-6 are magic entries for cases where the autopilot is implementing the camera manager itself.